extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/mediaModels/GoogleMediaItem.java

```diff
@@ -107,7 +114,23 @@ public static PhotoModel convertToPhotoModel(
         albumId.orElse(null),
         false  /*inTempStore*/,
         null  /*sha1*/,
-        mediaItem.getUploadedTime());
+        getCreationTime(mediaItem));
+  }
+
+  private static Date getCreationTime(GoogleMediaItem mediaItem) throws Exception  {
+    if(Strings.isNullOrEmpty(mediaItem.getMediaMetadata().getCreationTime()))
+    {
+      throw new NullPointerException("Skipping item transfer : creationTime field in media item was found to be null. This"
+          + " should never happen.");
+    }
+    try {
+      return CREATION_TIME_FORMAT.parse(mediaItem.getMediaMetadata().getCreationTime());
+    }
+    catch (ParseException parseException)
+    {
+      throw new IllegalArgumentException("Skipping item transfer since code failed to parse the string to get creationTime. "
+          + "Let's look into the dataFormatter");
+    }
```

Comment:
Let's create a new exception for this, and then change the method signature to throw this specific exception. 
```suggestion
    try {
      return CREATION_TIME_FORMAT.parse(mediaItem.getMediaMetadata().getCreationTime());
    }
    catch (ParseException parseException)
    {
      throw new FailedToParseDateException("Creation time failed to be parsed", parseException);
    }
```

Reply:
And maybe create an Exception interface called NonFatalException which FailedToParseDateException would implement. That way its really clear in code what we've decided

Reply:
but also feel free to not do this, we can add later